### PR TITLE
[MaxTwin] prepare distill pipeline for xm use

### PR DESCRIPTION
**Pull request by vlad-karp**
# Description

This PR updates the distillation trainer to support execution via XManager (XM) and aligns its initialization logic with other standard MaxText trainers 

Why is this change being made: To enable distillation workloads to run effectively within the XManager and Pathways ecosystems, ensuring consistent environment setup and performance optimizations ([ source](https://github.com/AI-Hypercomputer/maxtext/compare/vladk/maxtwin_xm_support?expand=1&content_ref=pathwaysutils+initialize)).  
Problem being solved: The distillation pipeline previously lacked explicit Pathways initialization and specific JAX/XLA configurations required for robust performance on TPU-based clusters ([ source](https://github.com/AI-Hypercomputer/maxtext/compare/vladk/maxtwin_xm_support?expand=1&content_ref=os+environ+libtpu_init_args+os+environ+get+libtpu_init_args+xla_tpu_spmd_rng_bit_generator_unsafe+true)).  

Implementation details:
- Copied Borg-required calls to pathwaysutils.initialize() and some initialization logic from other Maxtext trainers.
- A number of type hint fixes in distillation_utils.py to be able to build the package internally with blaze - Made student_config a required parameter in MaxTextCheckpointManager.

# Tests

The real test will be the internal xm launcher for the distill training run - to be patched.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/trainers/post_train/distillation/train_distill.py [hunk 1]
```diff
@@ -873,6 +876,14 @@ def main(argv: Sequence[str]) -> None:
   Args:
     argv: List of command-line arguments. Expects [script_name, config_file, ...].
   """
+
+  pathwaysutils.initialize()
+  jax.config.update("jax_default_prng_impl", "unsafe_rbg")
+  if "xla_tpu_spmd_rng_bit_generator_unsafe" not in os.environ.get("LIBTPU_INIT_ARGS", ""):
+    os.environ["LIBTPU_INIT_ARGS"] = (
+        os.environ.get("LIBTPU_INIT_ARGS", "") + " --xla_tpu_spmd_rng_bit_generator_unsafe=true"
+    )
```
Comment (gagika) [inline src/maxtext/trainers/post_train/distillation/train_distill.py:885]: should we add this under a flag?

Comment (vlad-karp) [reply src/maxtext/trainers/post_train/distillation/train_distill.py:885]: @gagika , it is common snippet from other maxtext trainers and they are not guarded by any flags.
Example: https://github.com/AI-Hypercomputer/maxtext/blob/be4fd71c8f114bfcbba98572b21cd53f06fa19c2/src/maxtext/trainers/pre_train/train.py#L787-L795

pathwaysutils.initialize() is a no-op for our current xpk/gcp training

**Review by JamesDeng42 — APPROVED** [review]
LGTM

**Review by gagika — APPROVED** [review]

File: src/maxtext/trainers/post_train/distillation/train_distill.py (hunk 1 above)
Comment (entrpn) [reply src/maxtext/trainers/post_train/distillation/train_distill.py:885]: +1 on the flag. is pathwaysutils.initialize() safe in xpk/xm?

Comment (vlad-karp) [reply src/maxtext/trainers/post_train/distillation/train_distill.py:885]: > +1 on the flag. is pathwaysutils.initialize() safe in xpk/xm?

@entrpn , yes, It is a no-op for Standard non-pathways workloads - you can see this code line in every maxtext trainer script.
